src/clickonce/launcher/Program.cs

```diff
@@ -126,6 +130,32 @@ private static void StartLoggingIfConfigured()
             }
         }
 
+        /// <summary>
+        /// Add ClickOnce properties from ApplicationDeployment object
+        /// as environment variables, to be passed to child process.
+        /// 
+        /// Add some extra variables, like Launcher version.
+        /// </summary>
+        private static void AddClickOnceEnvironmentVariables()
+        {
+            if (ApplicationDeployment.IsNetworkDeployed)
+            {
+                ApplicationDeployment ad = ApplicationDeployment.CurrentDeployment;
+
+                Environment.SetEnvironmentVariable("ClickOnce_ActivationUri", ad.ActivationUri?.ToString());
+                Environment.SetEnvironmentVariable("ClickOnce_CurrentVersion", ad.CurrentVersion?.ToString());
+                Environment.SetEnvironmentVariable("ClickOnce_DataDirectory", ad.DataDirectory?.ToString());
+                Environment.SetEnvironmentVariable("ClickOnce_IsFirstRun", ad.IsFirstRun.ToString());
+                Environment.SetEnvironmentVariable("ClickOnce_TimeOfLastUpdateCheck", ad.TimeOfLastUpdateCheck.ToString());
```

Comment:
I'd suggest using the round-trip format (O) rather than the default culture-specific format to preserve timezone and sub-seconds (assuming they are present).